arch/Kconfig

```diff
@@ -902,6 +907,16 @@ config ICACHE
 	help
 	  This option enables the support for the instruction cache (i-cache).
 
+config CACHE_COHERENCY
```

Comment:
This name likewise seems not to be capturing the issue at hand, which is the double-mapping behavior where a pointer can be cheaply converted to point to the same cached/uncached memory at different locations.  Also it's sort of colliding with CONFIG_KERNEL_COHERENCE, which gates the kernel support for feature (and also has the same kind of ambiguity).

Maybe call it "CACHEMAP" or "CACHE_DOUBLEMAP" or something?